lib/internal/buffer_new.js

```diff
@@ -103,7 +103,10 @@ function fromObject(obj) {
     return b;
   }
 
-  // TODO(trevnorris): This will fail for an actual ArrayBuffer.
+  if (obj instanceof ArrayBuffer) {
```

Comment:
I am working on a patch for master branch that will throw instead of abort if someone does something foolish like that. Though I should probably just add the check at the native layer here because it's an exception.